kms/src/test/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/InMemoryEdekTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+class InMemoryEdekTest {
+
+    @Test
+    void testEqualsAndHashCode() {
+        var edek1 = new InMemoryEdek(1, new byte[]{ (byte) 1, (byte) 2, (byte) 3 },
+                new byte[]{ (byte) 4, (byte) 5, (byte) 6 });
+
+        var edek2 = new InMemoryEdek(1, new byte[]{ (byte) 1, (byte) 2, (byte) 3 },
+                new byte[]{ (byte) 4, (byte) 5, (byte) 6 });
+
+        var edek3 = new InMemoryEdek(1, new byte[]{ (byte) 4, (byte) 5, (byte) 6 },
+                new byte[]{ (byte) 1, (byte) 2, (byte) 3 });
+
+        assertEquals(edek1, edek1);
```

Comment:
aside: we did say that we'd favour the more expressive assertj's `assertThat`.  We've been using that in new test code..

Reply:
I'll change it if you insist, but it doesn't seem like it adds anything of value.